### PR TITLE
Implement Series.rename for iterable

### PR DESCRIPTION
**Pull request by phofl**

### CODE REVIEW

File: dask_expr/_expr.py [hunk 1]
```diff
@@ -1041,10 +1042,34 @@ def _simplify_up(self, parent):
...
+
+    def _divisions(self):
+        index = self.operand("index")
+        if is_scalar(index) and not isinstance(index, Callable):
+            return self.frame.divisions
+        elif self.sorted_index and self.frame.known_divisions:
+            old = pd.Series(1, index=self.frame.divisions)
+            new_divisions = old.rename(index).index
+            if not new_divisions.is_monotonic_increasing:
+                raise ValueError("new divisions are not monotonic")
```
[the remark this answers is not in the document]
Comment (phofl) [reply dask_expr/_expr.py]: updated

File: dask_expr/_expr.py [hunk 2]
```diff
@@ -1041,10 +1042,37 @@ def _simplify_up(self, parent):
...
+    def _divisions(self):
+        index = self.operand("index")
+        if is_scalar(index) and not isinstance(index, Callable):
+            return self.frame.divisions
+        elif self.sorted_index and self.frame.known_divisions:
+            old = pd.Series(1, index=self.frame.divisions)
+            new_divisions = old.rename(index).index
+            if not new_divisions.is_monotonic_increasing:
+                raise ValueError(
+                    "The renamer creates an Index with non-monotonic divisions. "
```
Comment (milesgranger) [inline dask_expr/_expr.py:1069]:
```suggestion
                    "The rename creates an Index with non-monotonic divisions. "
```

Comment (phofl) [reply dask_expr/_expr.py:1069]: No that was intentional (it might not be proper English though?), renamer as the renaming function

Comment (milesgranger) [reply dask_expr/_expr.py:1069]: ah, okay. I thought it could also work like that. :+1:

**Pull request merged**